owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/main/java/fr/openwide/core/wicket/more/util/visit/VisitFilters.java

```diff
@@ -0,0 +1,226 @@
+	}
+
+	public static IVisitFilter downToIncluding(Object object) {
+		return downToIncluding(Object.class, Predicates.equalTo(object));
+	}
+	
+	public static IVisitFilter every(IVisitFilter ... operands) {
+		return new EveryVisitFilter(Arrays.asList(operands));
+	}
+	
+	public static IVisitFilter every(Iterable<? extends IVisitFilter> operands) {
+		return new EveryVisitFilter(operands);
+	}
+	
+	private static final class EveryVisitFilter implements IVisitFilter, Serializable {
+		private static final long serialVersionUID = 1L;
+		
+		private final Iterable<? extends IVisitFilter> operands;
+
+		public EveryVisitFilter(Iterable<? extends IVisitFilter> operands) {
+			super();
+			this.operands = operands;
+		}
+
+		@Override
+		public boolean visitObject(Object object) {
+			for (IVisitFilter filter : operands) {
+				if (!filter.visitObject(object)) {
+					return false;
+				}
+			}
+			return true;
+		}
+
+		@Override
+		public boolean visitChildren(Object object) {
+			for (IVisitFilter filter : operands) {
+				if (!filter.visitChildren(object)) {
+					return false;
+				}
+			}
+			return true;
+		}
+	}
+	
+	public static IVisitFilter any(IVisitFilter ... operands) {
+		return new AnyVisitFilter(Arrays.asList(operands));
+	}
+	
+	public static IVisitFilter any(Iterable<? extends IVisitFilter> operands) {
+		return new AnyVisitFilter(operands);
+	}
+	
+	private static final class AnyVisitFilter implements IVisitFilter, Serializable {
+		private static final long serialVersionUID = 1L;
+		
+		private final Iterable<? extends IVisitFilter> operands;
+
+		public AnyVisitFilter(Iterable<? extends IVisitFilter> operands) {
+			super();
+			this.operands = operands;
+		}
+
+		@Override
+		public boolean visitObject(Object object) {
+			for (IVisitFilter filter : operands) {
+				if (filter.visitObject(object)) {
+					return true;
+				}
+			}
+			return false;
+		}
+
+		@Override
+		public boolean visitChildren(Object object) {
+			for (IVisitFilter filter : operands) {
+				if (filter.visitChildren(object)) {
+					return true;
+				}
+			}
+			return false;
+		}
+	}
+
+	public static IVisitFilter renderedComponents() {
+		return RenderedComponentsVisitFilter.INSTANCE;
+	}
+	
+	private enum RenderedComponentsVisitFilter implements IVisitFilter {
+		INSTANCE;
+
+		@Override
+		public boolean visitObject(Object object) {
+			return hasRenderedMarkup(object);
+		}
+
+		private boolean hasRenderedMarkup(Object object) {
+			if (!(object instanceof Component)) {
+				return false;
+			}
+			Component component = (Component) object;
+			if (!component.hasBeenRendered() && !component.getOutputMarkupPlaceholderTag()) {
+				return false;
+			}
+			Component parent = component.getParent();
+			return component.isVisibleInHierarchy()
+					|| (parent == null || parent.isVisibleInHierarchy()) && component.getOutputMarkupPlaceholderTag();
+		}
+
+		@Override
+		public boolean visitChildren(Object object) {
+			return hasRenderedMarkup(object);
```

Comment:
No, it's fine ; we don't use a visitor ; we just use visitObject method, so there is no hierarchy traversal. Fine.